torch/distributed/fsdp/flat_param.py

```diff
@@ -755,7 +816,21 @@ def _use_unsharded_flat_param(
         unsharded_size = self.flat_param._unpadded_unsharded_size
         self.flat_param.data = padded_unsharded_flat_param[
             : unsharded_size.numel()
-        ].view(unsharded_size)
+        ].view(
+            unsharded_size
+        )  # this `.view()` is not autograd visible
+        # TODO (awgu): For `use_orig_params=True`, we create the unsharded
+        # views here, not through the FPW. This enables a *single* place for
+        # unsharded view creation during runtime. We should coalesce the
+        # `use_orig_params=False` code path to do the same.
+        if self._use_orig_params:
+            if self._training_state != HandleTrainingState.FORWARD:
+                # NOTE: `as_params=True` suffices here because we only need to
+                # restore the tensor *values* for backward computation. It
+                # does not need to be a fresh `Tensor` view.
+                self._use_unsharded_views(as_params=True)
+            else:
+                self._use_unsharded_views(as_params=False)
```

Comment:
simplify this to `self._use_unsharded_views(as_params=(self._training_state != FORWARD))`?

Reply:
Hmm, maybe we keep it like this so that the comment is more visible. It does not matter matter either way.